chameleonultragui/lib/sharedprefsprovider.dart

```diff
@@ -78,22 +78,67 @@ class CardSave {
 
   factory CardSave.fromJson(String json) {
     Map<String, dynamic> data = jsonDecode(json);
-    final id = data['id'] as String;
-    final uid = data['uid'] as String;
-    final sak = data['sak'] as int;
-    final atqa = List<int>.from(data['atqa'] as List<dynamic>);
-    final ats = List<int>.from((data['ats'] ?? []) as List<dynamic>);
-    final name = data['name'] as String;
-    final tag = getTagTypeByValue(data['tag']);
-    if (data['color'] == null) {
-      data['color'] = colorToHex(Colors.deepOrange);
-    }
-    final color = hexToColor(data['color']);
-    final encodedData = data['data'] as List<dynamic>;
+
+    final String id;
+    final String uid;
+    final int sak;
+    final List<int> atqa;
+    final List<int> ats;
+    final String name;
+    final TagType tag;
+    final Color color;
     List<Uint8List> tagData = [];
-    for (var block in encodedData) {
-      tagData.add(Uint8List.fromList(List<int>.from(block)));
+
+    if (data['Created'] == "proxmark3") {
```

Comment:
big no, this is cardsave, intended only for OUR card saves. Implement converter somewhere outside

Reply:
pm3 json is json, so the function fromjson should handle whatever json you throw at it. And adding a check like that in the import function would require looking at the json and then we try to unnescesarely parse json